packages/golden_toolkit/lib/src/testing_tools.dart

```diff
@@ -164,5 +165,9 @@ Future<void> screenMatchesGolden(
   );
 }
 
+// Matches Golden file is the easiest way for the images to be requested.
+Future<void> _primeImages(String fileName, Finder finder) =>
+    matchesGoldenFile(fileName).matchAsync(finder);
+
```

Comment:
what is the purpose of extracting it out?

Reply:
Corey wanted a comment around why this matchesGoldenFile was here.  Moving it out to a function seemed to make it more explicit

Reply:
@mZadorskii in reading some of the flutter bugs about this issue, there may be more intuitive & accurate ways to achieve this. 

I *believe* firing off this matcher currently works for two reasons:
1. it triggers the render cycle and this triggers the async image codec calls to get made
2. it is expensive, so it (most of the time) provides enough of a time buffer that images have been resolved before the subsequent golden assertion runs.

However, I think it is less than optimal. It's true intent isn't clear (hence this extraction), and I think it is *more* expensive than it needs to be. It appears there are some viable, but imperfect, workarounds that handle the vast majority of cases, by using Finders to find all Image widgets and synchronously pre-cache the images. If that strategy works, it would guarantee that images resolve before capturing the golden.

I want to experiment with this when we have some time.